tensorflow/python/ops/image_ops_impl.py

```diff
@@ -1324,26 +1324,32 @@ def decode_image(contents, channels=None, name=None):
       raise ValueError('channels must be in (None, 0, 1, 3)')
     substr = string_ops.substr(contents, 0, 4)
 
+    def _bmp():
+      is_bmp = math_ops.equal(substr, 'BM', name='is_bmp')
+      decode_msg = 'Unable to decode bytes as JPEG, PNG, GIF, or BMP'
+      assert_decode = control_flow_ops.Assert(is_bmp, [decode_msg])
```

Comment:
This is unused, you probably want to use a control dependency like in the other functions